### ISSUE TITLE
Can We Evaluate Domain Adaptation Models Without Target-Domain Labels?

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#50]
### Paper Link
https://openreview.net/forum?id=fszrlQ2DuP
## Task Description
**Target-domain labels** that are usually **unavailable** in real-world scenarios. As a result, it can be difficult to determine the effectiveness of UDA methods and how well they perform in the target domain.

Find an unsupervised metric for UDA model evaluation and selection, addressing three crucial challenges in real-world UDA scenarios: 
- selecting the **best UDA model** from a range of UDA method candidates; 
- adjusting **hyperparameters** to prevent the negative transfer and achieve enhanced performance; and 
- identifying the **optimal checkpoint** (epoch) of model parameters to avoid overfitting.

### Uniformity

- A classifier trained on the source domain often exhibits **biased predictions** when applied to the target domain. This bias stems from an **over-emphasis on certain classes** in the source domain due to their larger number of samples (Jamal et al., 2020).
- Consequently, we hypothesize that **a classifier with superior transferability should divide the feature space evenly and generate class-balanced prediction, rather than disproportionately emphasizing specific classes.**
- Prior theoretical research has also demonstrated that **evenly partitioning the feature space** leads to improved model generalization ability (Wang et al., 2020).

<img width="1270" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/a0c399a6-e4af-4780-8abf-aad8a72d4847">

### Hopkins Statistic 
- The Hopkins statistic assesses **whether the data is uniformly and randomly distributed** and measures the **clarity of the clusters**.
- For a **good UDA model**, the feature space should exhibit **distinct clusters for each class**, indicating better transferability and discriminability.
- We assume that the **target domain accuracy should be correlated with the Hopkins statistic**.

<img width="1250" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/389f2217-1cb9-4adf-9884-75df62d8a917">

- However, **all samples form a single cluster or the classifier boundary intersects a densely populated region of samples**, the Hopkins statistic can still yield a **high value**.
- Thus, by incorporating **mutual information**, we can discern the prediction confidence and diversity (class balance) of the UDA model.

<img width="1256" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/250949ed-832a-41d4-9051-1236c6731ff1">

### Transfer Score
<img width="1269" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/daffa14c-47ae-43d7-86b8-f6a2cf6c1fdc">

#### Saturation Level of UDA Training.
- Due to potential overfitting, the highest target-domain accuracy is often achieved during the training process.
- To determine the optimal epoch for model selection after UDA, we introduce the saturation level of UDA training, represented by the coefficient of variation of the TS.

<img width="1252" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/a80ebe8b-d3ac-4b42-8401-97844933fbea">

- When the saturation level of the TS falls below a predefined threshold ζ , it indicates that the TS has reached a point of saturation.
- Beyond this threshold, training the model further could potentially result in a decline in performance.
- To determine the optimal checkpoint, we select the epoch with the highest TS within that time window.

## Experiments 
<img width="1645" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/5dcada3b-ae4a-42db-8470-048fc42907a5">

### Task 1: UDA Method Selection. 

<img width="1717" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/000490a5-80ab-4bf0-bfee-94ec4c1a6b52">
 
### Task 2: Hyperparameter Tuning. 

<img width="1700" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/4d391512-ab11-4edf-9667-feb8b4ce5b3c">

### Task 3: Choose a Good Checkpoint. 

<img width="1644" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/0cf75dc9-b031-450e-9f82-418a89ace276">

<img width="1055" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/60a63e2c-af73-40e6-9967-46313c48d2b7">
